### ISSUE TITLE
clusterResources skips several parts for GKE clusters, without namespace specified

### ISSUE THREAD

**Issue by xavpaice** [replicatedhq/troubleshoot#1167]
**Bug Description**

I have a GKE cluster, with the ability to run `kubectl get pods` on all namespaces (plus other permissions).

I expect to see, in a namespace that has deployments, pods, etc to see those listed.  I do not, I see a very minimal set of information.

If I specify the namespace, everything works as expected.

**Expected Behavior**

Given that my user has access to the resources required to collect from all namespaces, I expect to see information including deployments etc for all namespaces unless I specify otherwise.

**Steps To Reproduce**

Include the commands to reproduce the issue including any output. Any information that will help us to understand the problem is useful.  Feel free to paste long output into a [Github gist](https://gist.github.com) and include the link here.

* Requires a GKE cluster with multiple namespaces

Run `support-bundle` using the following minimal spec:

```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: sample
spec:
  collectors:
    - clusterResources: {}
```

**Additional Context**

https://github.com/replicatedhq/troubleshoot/blob/6de79afc35a514cd65294403f939a814e00c065b/pkg/collect/cluster_resources.go#L1488 returns only one ResourceRules for the namespace, with the Verb "create".  The same exercise against a local k3d cluster has an extra ResourceRules entry with Verb "*" for the namespace, and Resources "*".

This causes https://github.com/replicatedhq/troubleshoot/blob/6de79afc35a514cd65294403f939a814e00c065b/pkg/collect/cluster_resources.go#L1542 to return false for every namespace, thus very little collection happens.

**Comment by banjoh** [comment]
There are a few ways we can address this bug.
* Stop checking RBAC permissions and have troubleshoot take a best-effort strategy in collecting cluster resources. Record RBAC errors in the support bundle just like any other errors. **NOTE:** This would be breaking change cause of the [ignoreRBAC](https://github.com/replicatedhq/troubleshoot/blob/main/pkg/apis/troubleshoot/v1beta2/collector_shared.go#L27) parameter. The parameter is not [documented](https://troubleshoot.sh/docs/collect/cluster-resources/#parameters) so its probably not used.
* Improve how we evaluate RBAC permissions [here](https://github.com/replicatedhq/troubleshoot/blob/main/pkg/collect/cluster_resources.go#L1611) and change [ignoreRBAC](https://github.com/replicatedhq/troubleshoot/blob/main/pkg/apis/troubleshoot/v1beta2/collector_shared.go#L27) default to `true`.
* Document (https://github.com/replicatedhq/troubleshoot/blob/main/pkg/collect/cluster_resources.go#L1611) and change [ignoreRBAC](https://github.com/replicatedhq/troubleshoot/blob/main/pkg/apis/troubleshoot/v1beta2/collector_shared.go#L27) to have users know about it. We would still need to improve our RBAC implementation to at least catch all default k8s resources as opposed to just [collecting pods](https://github.com/replicatedhq/troubleshoot/blob/main/pkg/collect/cluster_resources.go#L1633)

**Comment by banjoh** [comment]
FYI, I managed to reproduce this bug with in a GKE cluster

**Comment by adamancini** [comment]
@banjoh so was I and adding `ignoreRBAC: true` to the spec caused it to succeed:

```
 ✔  alice  ~/go/src/github.com/replicatedhq/troubleshoot/support-bundle-2023-06-06T12_33_03/cluster-resources   watch ● ?  tree -d  
.
├── auth-cani-list
├── cronjobs
├── custom-resources
│   ├── ciliumendpoints.cilium.io
│   └── redirectservices.networking.gke.io
├── deployments
├── events
├── image-pull-secrets
│   ├── annarchy
│   ├── default
│   ├── haic
│   ├── kube-node-lease
│   └── kube-public
├── ingress
├── jobs
├── limitranges
├── network-policy
├── pod-disruption-budgets
├── pods
│   └── logs
│       └── default
│           ├── foo-config-service-58bb5868bb-6wcdf
│           ├── foo-data-catalog-7b5ff8884d-gqpf6
│           ├── foo-orch-5f49585bfd-47rzw
...
```

**Comment by adamancini** [comment]
from a GKE Autopilot cluster during [`getSelfSubjectRulesReviews`](https://github.com/adamancini/troubleshoot/blob/3ef3733e2c1fc28dcec58c3ad0c48a800e5ccef7/pkg/collect/cluster_resources.go#L1540) we end up with a response like this trying to call kubernetes `client.AuthorizationV1().SelfSubjectRulesReviews()`:

```
 "annarchy": {
    "resourceRules": [
      {
        "verbs": [
          "create"
        ],
        "apiGroups": [
          "authorization.k8s.io"
        ],
        "resources": [
          "selfsubjectaccessreviews",
          "selfsubjectrulesreviews"
        ]
      }
    ],
    "nonResourceRules": [
      {
        "verbs": [
          "get"
        ],
        "nonResourceURLs": [
          "/healthz",
          "/livez",
          "/readyz",
          "/version",
          "/version/"
        ]
      },
      {
        "verbs": [
          "get"
        ],
        "nonResourceURLs": [
          "/api",
          "/api/*",
          "/apis",
          "/apis/*",
          "/healthz",
          "/livez",
          "/openapi",
          "/openapi/*",
          "/readyz",
          "/version",
          "/version/"
        ]
      }
    ],
    "incomplete": true,
    "evaluationError": "webhook authorizer does not support user rule resolution"
```

and that error gets returned from https://github.com/kubernetes/apiserver/blob/master/plugin/pkg/authorizer/webhook/webhook.go#LL249C5-L249C5

**Comment by adamancini** [comment]
and it doesn't look like support for the webhook authorizers has had any traction in upstream https://github.com/kubernetes/kubernetes/issues/52745

**Comment by swhi3635** [comment]
I was encountering this issue on a GKE cluster as well.  Serving an affected support bundle with `sbctl` resulted in the following error when running `kubectl get pods`:
```
Error from server (NotFound): Unable to list "/v1, Resource=pods": the server could not find the requested resource (get pods)
```
In the sbctl logs, I saw:
```
INFO[0033] failed to load file open /var/folders/fd/ggszn1nn3ds2nyztw14plwdm0000gp/T/sbctl-3808171838/support-bundle-2023-06-07T22_01_09/cluster-resources/pods/default.json: no such file or directory
```

The problem was resolved by adding `ignoreRBAC: true` to the collector, as mentioned above:
```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: foo
spec:
  collectors:
    - clusterResources:
        ignoreRBAC: true 
```

I had a bit of trouble finding this issue initially, so hopefully the attached logs will help others find this if they're having the same problem.  Thanks for the quick work on this!

**Comment by xavpaice** [comment]
@adamancini other than the docs fix (thanks, merged!), is there anything we can do to resolve this automatically?  Maybe, if we get the message you noted can we retry with the `ignoreRBAC` setting - afaik it was only put there in order to reduce noise if someone doesn't have permissions, rather than stop the attempt entirely.

**Comment by adamancini** [comment]
@xavpaice yeah I think we should definitely surface a good warning to the user at a minimum; I think I'm in favor of falling back to trying `ignoreRBAC` if we catch this particular error

**Comment by xavpaice** [comment]
Implemented in #1243
